### DISCUSSION TITLE
Does fsspec provide a way to get a file's public cloud url?

### DISCUSSION THREAD

**Discussion by pedro-cf** [fsspec/filesystem_spec#1605] [Q&A]
Does fsspec provide a way to get a file's public cloud url?

**Comment by martindurant** [comment]
I'm not sure what you mean. Some filesystems provide fs.sign(), and all provide fs.unstrip_protocol() to make an fs-specific URL more general.

**Reply by pedro-cf** [reply]
f.e.

If I open my file on azure blob storage:

I would like be able to get it's respective public url:

https://my-bucket.blob.core.windows.net/test/test.txt

**Reply by martindurant** [reply]
The you need .sign(). If this isn't implemented, please ask on the adlfs repo.

**Reply by pedro-cf** [reply]
I don't mean a temporary access link, just the direct permanent link

**Reply by martindurant** [reply]
You should ask at adlfs
